test/testdata/dsl/t_struct_some_default.rb

```diff
@@ -0,0 +1,11 @@
+# typed: true
+
+class SomeDefault < T::Struct
+  prop :foo, Integer
+  prop :bar, T::Boolean, default: false
+end
+
+SomeDefault.new(foo: 1)
+SomeDefault.new(foo: 2, bar: true)
+SomeDefault.new(foo: 3, bar: false)
+SomeDefault.new # error: Missing required keyword argument `foo` for method `SomeDefault#initialize`
```

Comment:
These all seem like good simple test cases, but they don't really test edge-casey behaviors. From your experience codemodding pay-server, do you think there's any test you'd like to write to accommodate things that came up? Specifically around inheritance or the things you ended up having to `T.unsafe` or make into an `T::InexactStruct`.

Also, are there test that the non-T::Struct subclasses have untyped initialize methods? If not it would be good to document that behavior too.

Reply:
Though I did spend a few days going over pay-server to fix bugs surfaced by this change, I'm not sure if I can think of anything very "edge-casey". Most bugs seemed like fairly simple "this argument is not the right type" or had to do with overridden `initialize` which is not a pattern we want to support super well.

I can add a test that asserts that subclasses of `T::InexactStruct` don't get well-typed `new`.